cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillPreparer.java

```diff
@@ -508,9 +522,18 @@ public TwillController start(long timeout, TimeUnit timeoutUnit) {
       StatefulRunnable statefulRunnable = statefulRunnables.get(mainRuntimeSpec.getName());
       Type resourceType = statefulRunnable == null ? V1Deployment.class : V1StatefulSet.class;
 
+      String programType = getProgramTypeFromAppName(twillSpec.getName());
```

Comment:
this approach assumes special knowledge about CDAP program types, which this layer shouldn't really know about. 

I think it would be better if something further up the stack in app-fabric, which knows whether it's a workflow or a worker, sets some property that is then used to tell this class to use a Deployment or a Job.

Reply:
I see. Will move this part there.

Reply:
Done. Using withConfig option of TwillPreparer to pass in property information. This can be used to get more information about programs such as worker number of instances etc.